### PR TITLE
Generate Payments Report

### PR DESCRIPTION
**Pull request by jenlij**

### CODE REVIEW

File: cmd/payments/main.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	}
+)
+
+// Payments gives access to the payments api
+type Payments struct {
+	client pb.PaymentsClient
+}
+
+func main() {
+	rootCmd.PersistentFlags().StringVarP(&port, "port", "p", ":10000", "storj-sdk satellite port")
```
Comment (thepaul) [inline cmd/payments/main.go]: It's going to be storj-sim, now, I think :D

File: pkg/payments/server.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+	// TODO
+	return &pb.AdjustPricesResponse{}, PaymentsError.New("AdjustPrices not implemented")
+}
+
+// GenerateCSV creates a csv file for payment purposes
+func (srv *Server) GenerateCSV(ctx context.Context, req *pb.GenerateCSVRequest) (*pb.GenerateCSVResponse, error) {
+	fmt.Println("entering server generate csv")
+	start, err := ptypes.Timestamp(req.StartTime)
+	if err != nil {
+		return &pb.GenerateCSVResponse{}, PaymentsError.Wrap(err)
```
Comment (wthorp) [inline pkg/payments/server.go]: All these '&pb.GenerateCSVResponse{}' would be clearer as nil, yes?

File: cmd/payments/main.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+		StartTime: startTimestamp,
+		EndTime:   endTimestamp,
+	}
+
+	resp, err := p.client.GenerateCSV(ctx, req)
+	if err != nil {
+		return ErrRequest.Wrap(err)
+	}
+
+	fmt.Println("Created payments report at", resp.GetFilepath())
```
Comment (wthorp) [inline cmd/payments/main.go]: If 'resp.GetFilepath()' is a local path... why did we need a client/server architecture?

File: pkg/pb/payments.proto [hunk 4]
```diff
@@ -12,14 +13,17 @@ service Payments {
     rpc Calculate(CalculateRequest) returns (CalculateResponse);
     // AdjustPrices sets the prices paid by a satellite for data at rest and bandwidth
     rpc AdjustPrices(AdjustPricesRequest) returns (AdjustPricesResponse);
+        // GenerateCSV creates a csv file for payment purposes
+    rpc GenerateCSV(GenerateCSVRequest) returns (GenerateCSVResponse);
+    //TODO REMOVE
```
Comment (thepaul) [inline pkg/pb/payments.proto]: Should this be removed now? Or sometime in the future?

File: satellite/satellitedb/accounting.go [hunk 5]
```diff
@@ -175,3 +175,61 @@ func (db *accountingDB) SaveRollup(ctx context.Context, latestRollup time.Time,
 	_, err = tx.Update_AccountingTimestamps_By_Name(ctx, dbx.AccountingTimestamps_Name(accounting.LastRollup), update)
 	return Error.Wrap(err)
 }
+
+// QueryPaymentInfo queries StatDB, Accounting Rollup on nodeID
+func (db *accountingDB) QueryPaymentInfo(ctx context.Context, start time.Time, end time.Time) ([]*accounting.CSVRow, error) {
+	s := dbx.AccountingRollup_StartTime(start)
+	e := dbx.AccountingRollup_StartTime(end)
+	data, err := db.db.All_Node_Id_Node_CreatedAt_Node_AuditSuccessRatio_AccountingRollup_StartTime_AccountingRollup_PutTotal_AccountingRollup_GetTotal_AccountingRollup_GetAuditTotal_AccountingRollup_GetRepairTotal_AccountingRollup_PutRepairTotal_AccountingRollup_AtRestTotal_By_AccountingRollup_StartTime_GreaterOrEqual_And_AccountingRollup_StartTime_Less_OrderBy_Asc_Node_Id(ctx, s, e)
```
Comment (thepaul) [inline satellite/satellitedb/accounting.go]: lol this is amazing

File: pkg/pb/payments.proto (hunk 4 above)
Comment (jenlij) [reply pkg/pb/payments.proto]: will remove in ticket https://storjlabs.atlassian.net/browse/V3-1084 :)

File: cmd/payments/main.go (hunk 1 above)
Comment (jenlij) [reply cmd/payments/main.go]: awesome ty

File: cmd/payments/main.go (hunk 3 above)
Comment (jenlij) [reply cmd/payments/main.go]: true... perhaps we should change this down the line

File: cmd/payments/main.go [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+	}
+)
+
+// Payments gives access to the payments api
+type Payments struct {
+	client pb.PaymentsClient
+}
+
+func main() {
+	rootCmd.PersistentFlags().StringVarP(&port, "port", "p", ":10000", "storj-sim satellite port")
```
Comment (egonelbre) [inline cmd/payments/main.go]: "satellite port"

File: cmd/payments/main.go [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+// NewPayments creates a payments object
+func NewPayments() (*Payments, error) {
+	identity, err := provider.NewFullIdentity(ctx, 12, 4)
+	if err != nil {
+		return &Payments{}, ErrIdentity.Wrap(err)
+	}
+	tc := transport.NewClient(identity)
+	conn, err := tc.DialAddress(ctx, port)
```
Comment (egonelbre) [inline cmd/payments/main.go]: this might be blocked after requiring authorization... so use satellite identity or later private grpc

**Pull request merged**